src/pages/index.js

```diff
@@ -1,60 +1,123 @@
 import "./index.css";
-
+import Api from "../components/Api.js";
 import Card from "../components/Card.js";
 import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
-import {initialCards, selectors, validationSettings, profileEditForm, addCardForm, nameInput, aboutInput} from "../utils/constants.js";
+import { selectors, validationSettings, profileEditForm, addCardForm, editAvatarForm, nameInput, aboutInput } from "../utils/constants.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
+
+// Initialize api
+const api = new Api({baseUrl: "https://around-api.en.tripleten-services.com/v1", authToken: "9ef0145a-4550-4d8d-9dc2-252c6d573e29"});
+
+
+// create delete card popup
+const deleteCardPopup = new PopupWithConfirmation({
+  popupSelector: selectors.deleteCardPopup,
+  handleConfirm: (cardId) => {
+    api.deleteCard(cardId);
+  }
+});
 
 // Card renderer function
 const renderCard = (data) => {
   const cardElement = new Card(
     {
-      data,
+      data: data,
       handleImageClick: (imgData) => {
         cardPreviewPopup.open(imgData);
-      }
+      },
+      handleDelete: (card) => {
+        deleteCardPopup.open(card);
+        deleteCardPopup.setSubmitAction((card) => {
+          api.deleteCard(card.getId());
+          card.removeCard();
+        });
+      },
+      confirmPopup: deleteCardPopup,
+      api: api
     },
     selectors.cardTemplate
   );
   cardSection.addItem(cardElement.generateCard());
 }
 
 
+
+// Get initial user information and cards
+api.getUserInfo()
+  .then(res => {
+    document.querySelector(".profile__title").textContent = res.name;
+    document.querySelector(".profile__description").textContent = res.about;
+    document.querySelector(".profile__image").src = res.avatar;
+  });
+
+api.getInitialCards()
+  .then(res => {
+    res.reverse().forEach((card) => {
+      renderCard(card);
+    });
+  });
+
+
+
 // Create card and card image popup instances
 export const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
 export const cardSection = new Section({ renderer: renderCard }, selectors.cardList);
 
 
+
 // Create form and user info instances
 const userInfo = new UserInfo({nameSelector: selectors.nameElement, aboutMeSelector: selectors.descriptionElement});
 
 const editProfilePopup = new PopupWithForm(selectors.editProfilePopup, (data) => {
   userInfo.setUserInfo(data);
```

Comment:
`  userInfo.setUserInfo(data);
`

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change